tests/system/python/service/test_service_notification.py

```diff
@@ -0,0 +1,199 @@
+        def _add_notification_instance(payload):
+            conn = http.client.HTTPConnection(foglamp_url)
+            conn.request("POST", '/foglamp/notification', json.dumps(payload))
+            r = conn.getresponse()
+            assert 200 == r.status
+            r = r.read().decode()
+            jdoc = json.loads(r)
+            assert "Notification {} created successfully".format(payload['name']) == jdoc['result']
+
+        # FIXME: Handle in a better way; we need below code once for a test
+        if data['name'] == 'Test1':
+            self._install_notify_plugin(notify_branch)
+            remove_directories("/tmp/foglamp-notify-{}".format(NOTIFY_PLUGIN))
+
+        _add_notification_instance(data)
+
+    def test_delivery_rule_plugin_with_python35(self, foglamp_url):
+        jdoc = self._get_result(foglamp_url, '/foglamp/notification/plugin')
+        assert 1 == len(jdoc['delivery'])
+        assert NOTIFY_PLUGIN == jdoc['delivery'][0]['name']
+        assert 1 == len(jdoc['rules'])
+        assert NOTIFY_RULE == jdoc['rules'][0]['name']
+
+    def test_get_notifications_and_audit_entry(self, foglamp_url):
+        jdoc = self._get_result(foglamp_url, '/foglamp/notification')
+        assert 3 == len(jdoc['notifications'])
+
+        jdoc = self._get_result(foglamp_url, '/foglamp/audit?source=NTFAD')
+        assert 3 == jdoc['totalCount']
+
+    def test_update_notification(self, foglamp_url, name="Test1"):
+        conn = http.client.HTTPConnection(foglamp_url)
+        data = {"notification_type": "toggled"}
+        conn.request("PUT", '/foglamp/notification/{}'.format(name), json.dumps(data))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert "Notification {} updated successfully".format(name) == jdoc["result"]
+
+        # Verify updated notification info
+        jdoc = self._get_result(foglamp_url, '/foglamp/notification/{}'.format(name))
+        assert "toggled" == jdoc['notification']['notificationType']
+
+    def test_delete_notification(self, foglamp_url, name="Test3"):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("DELETE", '/foglamp/notification/{}'.format(name))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert "Notification {} deleted successfully.".format(name) == jdoc["result"]
+
+        # Verify only two notifications should exist NOT 3
+        jdoc = self._get_result(foglamp_url, '/foglamp/notification')
+        notifications = jdoc['notifications']
+        assert 2 == len(notifications)
+        assert "Notification Test1" == notifications[0]['name']
+        assert "Notification Test2" == notifications[1]['name']
+
+    @pytest.mark.skip(reason='To be Implemented')
+    def test_receive_notification(self, foglamp_url):
```

Comment:
do we some way to check/test foglamp instance for sent_notification? receivers are external we should stub them for now and ensure stub is called, with sent test only